emacs25Macport cannot use AppKit

## Issue description

It looks like emacs is not able to use Appkit for some reason. Here is the relevant logs:

```
checking AppKit/AppKit.h usability... no
checking AppKit/AppKit.h presence... yes
configure: WARNING: AppKit/AppKit.h: present but cannot be compiled
configure: WARNING: AppKit/AppKit.h:     check for missing prerequisite headers?
configure: WARNING: AppKit/AppKit.h: see the Autoconf documentation
configure: WARNING: AppKit/AppKit.h:     section "Present But Cannot Be Compiled"
configure: WARNING: AppKit/AppKit.h: proceeding with the compiler's result
configure: WARNING:     ## ------------------------------------ ##
configure: WARNING:     ## Report this to bug-gnu-emacs@gnu.org ##
configure: WARNING:     ## ------------------------------------ ##
checking for AppKit/AppKit.h... no
configure: error: The include files (AppKit/AppKit.h etc) that
are required for a Nextstep build are missing or cannot be compiled.
```

```
$ nix-shell --pure --run "echo '#import <AppKit/AppKit.h>' | cc -x objective-c-header -" -p stdenv.cc darwin.apple_sdk.frameworks.AppKit
In file included from <stdin>:1:
In file included from /nix/store/6zrda57fmgvn9yjpqfv0lm97vabma0f5-apple-framework-AppKit/Library/Frameworks/AppKit.framework/Headers/AppKit.h:10:
In file included from /nix/store/7f0bbz6lclkbmgr44n74kqkwq37p4dq5-apple-framework-Foundation/Library/Frameworks/Foundation.framework/Headers/Foundation.h:141:
In file included from /nix/store/7f0bbz6lclkbmgr44n74kqkwq37p4dq5-apple-framework-Foundation/Library/Frameworks/Foundation.framework/Headers/NSAppleEventDescriptor.h:7:
In file included from /nix/store/2qbqqy62d750c2gcq2pp94skqwapdg9n-apple-framework-ApplicationServices/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:39:
In file included from /nix/store/y5v7lnz3fiksjha0bij50zdbalvykqmc-apple-framework-CoreText/Library/Frameworks/CoreText.framework/Headers/CoreText.h:27:
In file included from /nix/store/y5v7lnz3fiksjha0bij50zdbalvykqmc-apple-framework-CoreText/Library/Frameworks/CoreText.framework/Headers/CTFramesetter.h:22:
In file included from /nix/store/y5v7lnz3fiksjha0bij50zdbalvykqmc-apple-framework-CoreText/Library/Frameworks/CoreText.framework/Headers/CTTypesetter.h:21:
/nix/store/y5v7lnz3fiksjha0bij50zdbalvykqmc-apple-framework-CoreText/Library/Frameworks/CoreText.framework/Headers/CTLine.h:22:10: fatal error: 'CoreFoundation/CFAttributedString.h' file not found
#include <CoreFoundation/CFAttributedString.h>
         ^
1 error generated.
```

It looks like Foundation framework is busted. This is lowPrio so I don't think there is Hydra logs.

nixpkgs commit: b69f568
nix version: 1.11.4

cc @copumpkin

--- a/pkgs/applications/editors/emacs/macport-24.5.nix
+++ b/pkgs/applications/editors/emacs/macport-24.5.nix
@@ -47,7 +47,7 @@ stdenv.mkDerivation rec {
     "--enable-mac-app=$$out/Applications"
   ];
 
-  CFLAGS = "-O3 -DMAC_OS_X_VERSION_MAX_ALLOWED=1090";
+  CFLAGS = "-O3 -DMAC_OS_X_VERSION_MAX_ALLOWED=1090 -DMAC_OS_X_VERSION_MIN_REQUIRED=1090";
   LDFLAGS = "-O3 -L${ncurses.out}/lib";
 
   postInstall = ''

--- a/pkgs/applications/editors/emacs/macport-25.1.nix
+++ b/pkgs/applications/editors/emacs/macport-25.1.nix
@@ -5,7 +5,7 @@
 
 stdenv.mkDerivation rec {
   emacsName = "emacs-25.1";
-  name = "${emacsName}-mac-6.0";
+  name = "${emacsName}-mac-6.1";
 
   builder = ./builder.sh;
 
@@ -16,7 +16,7 @@ stdenv.mkDerivation rec {
 
   macportSrc = fetchurl {
     url = "ftp://ftp.math.s.chiba-u.ac.jp/emacs/${name}.tar.gz";
-    sha256 = "2f7a3fd826e6dea541ada04f4a1ff2903a87a1f736b89c5b90bf7bb820568e34";
+    sha256 = "1zwxh7zsvwcg221mpjh0dhpdas3j9mc5q92pprf8yljl7clqvg62";
   };
 
   enableParallelBuilding = true;

--- a/pkgs/applications/editors/emacs/macport-25.1.nix
+++ b/pkgs/applications/editors/emacs/macport-25.1.nix
@@ -48,7 +48,7 @@ stdenv.mkDerivation rec {
     "--enable-mac-app=$$out/Applications"
   ];
 
-  CFLAGS = "-O3 -DMAC_OS_X_VERSION_MAX_ALLOWED=1090";
+  CFLAGS = "-O3 -DMAC_OS_X_VERSION_MAX_ALLOWED=1090 -DMAC_OS_X_VERSION_MIN_REQUIRED=1090";
   LDFLAGS = "-O3 -L${ncurses.out}/lib";
 
   postInstall = ''

